src/Features/LanguageServer/Protocol/Handler/SemanticTokens/SemanticTokensRangeHandler.cs

```diff
@@ -117,13 +130,32 @@ public void Dispose()
         }
 
         private void OnLspSolutionChanged(object? sender, WorkspaceChangeEventArgs e)
-            => EnqueueSemanticTokenRefreshNotification();
+        {
+            if (e.DocumentId is not null && e.Kind is WorkspaceChangeKind.DocumentChanged)
```

Comment:
Should `DocumentAdded` be here too?

Reply:
Good point, that would probably help speed things up on solution open

Reply:
Actually, I tested it out just now and it seems like adding `WorkspaceChangeKind.DocumentAdded` doesn't affect things much. I'm thinking this is because when the event fires on solution open the range handler hasn't been instantiated yet? Nonetheless, this logic is intended to focus on the typing scenario (which is currently pretty laggy), so I think just having `DocumentChanged` works for the vast majority of cases.